bin/config.js

```diff
@@ -36,46 +28,41 @@ const prepareIcons = (input) => {
   return files
 }
 
-const prepareOutput = (output) => {
-  const result = {
-    ...defaultConfig.output,
-    icons: { ...defaultConfig.output.icons },
-  }
-
-  if (typeof output == 'string') result.path = output
-  else {
-    if (output.path) result.path = output.path
-    if (output.publicPath) result.publicPath = output.publicPath
-
-    if (typeof output.icons == 'boolean')
-      result.icons = { ...defaultConfig.output.icons }
-    else if (typeof output.icons == 'string') {
-      result.icons = {
-        files: /.svg$/.test(output.icons) ? null : output.icons,
-        sprite: appendExtension(output.icons, '.svg'),
-      }
-    } else result.icons = { ...output.icons }
-
-    if (result.icons.files)
-      result.icons.files = path.resolve(result.path, result.icons.files)
-    if (result.icons.sprite)
-      result.icons.sprite = path.resolve(result.path, result.icons.sprite)
-  }
-
-  return result
+const mergeAndPrepareConfigs = (...configs) => {
+  // Create absolute path for icons so they can be merged
+  configs.forEach(({ config }) => {
+    if ((config || {}).icons) config.icons = prepareIcons(config.icons)
+  })
+
+  const mergedConfig = merge(...configs)
+  const {
+    config: { output },
+  } = mergedConfig
+  if (output.icons.files)
+    output.icons.files = path.resolve(output.path, output.icons.files)
+
+  if (output.icons.sprite)
+    output.icons.sprite = path.resolve(output.path, output.icons.sprite)
+
+  return mergedConfig
 }
 
-const prepare = (config) => {
-  const icons = prepareIcons(config.icons)
-  const output = prepareOutput(config.output)
-  const svgo = { ...defaultConfig.svgo, ...config.svgo }
-  return { icons, output, svgo }
+const read = async (filepath) => {
+  const explorer = cosmiconfig('design')
+  // If filepath is not supplied we start search for config from project root,
+  // The reason is to avoid loading the default config.
+  // (/<project_root>/node_modules/@oacore/design/bin)
+  const mainProjectDirectory = path.resolve(__dirname, '../../../../')
```

Comment:
Is there no better option? Maybe in `process`, e.g. `cwd` or something.

Also, please check how cosmiconfig works. I think it starts to lookup from CWD.

Reply:
CWD is this path `/<project_root>/node_modules/@oacore/design/` so it's actually the same (one `../` less)

Reply:
I checked StackOveflow, [they recommend `require.main.filename`](https://stackoverflow.com/a/18721515/7205866). I would try it but I am not sure it would work when running via Webpack, Next and even NPM.

Reply:
It may be safe enough but it also assumes that you do not run the command from itself what is not true. Also, we did not ship any images before but we should now. This should be solved somehow, or it is solved already... 🤔 Maybe we need some ignore rules, I am not sure.

Reply:
You also may try some magic like `require.resolve('package.json')` or something similar. ESLint does this, so exploring their code it another option.

However, I still hope, that `cosmiconfig` resolves the config from the project root, from the correct one. Also, it should give you a list when you 'explore'.

Reply:
It doesn't it starts from /<project_root>/node_modules/@oacore/design/ and goes one level up till it find config or it ends in home directory. So when I added config to this repo it always found this one and stopped the search.

So I would say we shouldn't use search at all. We should always try to find always only one config in the project root. And you are right about running the command from itself. I forgot to cover this case. I will have a look at the answer you provided. I think require.main.filename will be safe to use in this scenario.

Reply:
> So when I added config to this repo it always found this one and stopped the search.

Yes, but it accepts `searchFrom`. So we can find the first config, and recursively do `explorer.search(path.resolve('../', curerntConfig.filepath))` or similar. I still think it's better to keep `explorer.search()` since it does a lot of magic for us.

I was afraid that if we host the package differently, i.e. without `@oacore` prefix we will need to tweak the path.

When you link the package, the path it even more different, it's not under `project_root/node_modules/` but (in my case) along with the project as a sibling. It's another pain.

Reply:
I assume it can be just change like this and the problem is solved. It will work like before. We just skip the current config. I don't think it makes sense to support multiple configs. I don't see a use case for it. We should add it once it's needed/requested (as everything else).

``` 
const mainProjectDirectory = path.resolve(__dirname, '../../')
```

Reply:
In this case, I think that if you do a reduce loop anyway, why not to do it more flexible to prevent such requests? But I do not insist. If you want to keep it simple, go for it.

Reply:
Also, I don't think that cosmic config should search in user home directory. But I am not sure how to make it to stop at the certain directory. Anyway, I guess it like minor thing unless somebody complains 😄

Reply:
There is a [`stopDir` option](https://github.com/davidtheclark/cosmiconfig#stopdir) too. However, I am struggling to think the right path for it so I would agree with you 🙂

Reply:
Not sure what do you mean by reduce loop. Can you maybe share some code example?

Reply:
Sorry, I misled you and myself. Something like:

```js
const configs = []
let result
do {
  result = explorer.search(result?.filename)
  configs.push(result.config)
} while (result != null)
configs.pop() // `null` was added in the end

configs.reduce((result, current) => merge(result, current), defaultConfig)
```

but I messed up with paths, so you will need some extra conditions for handling it.

Reply:
I see. I don't think we have a use case for this. We just expect one config. So I think `explorer.search(path.resolve(__dirname, '../../'))` is more then enough.